### PR TITLE
Replication observability metrics

### PR DESCRIPTION
**Pull request by nitesh3108**
# PR Submission checklist
It enables API support for volume_mirror_transfer_rate_cma_view which will be helpful to pull the Replication related stats for Sync and Metro at volume level.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/1443|

# Common PR Checklist:

- [x] Have you made sure that the code compiles?
- [x] Have you commented your code, particularly in hard-to-understand areas
- [x] Did you run tests in a real Kubernetes cluster?
- [x] Have you maintained backward compatibility

## Description of your changes:

- Executed gopowerstore unit tests 

Earlier unit-test report
```
PASS
coverage: 82.2% of statements
ok      github.com/dell/gopowerstore    0.069s  coverage: 82.2% of statements
```

Current unit-test report
```
=== RUN   TestClientIMPL_VolumeMirrorTransferRate
--- PASS: TestClientIMPL_VolumeMirrorTransferRate (0.00s)

PASS
coverage: 83.8% of statements
ok      github.com/dell/gopowerstore    0.064s  coverage: 83.8% of statements
```

### CODE REVIEW

File: metrics.go [hunk 1]
```diff
@@ -45,6 +50,33 @@ func (c *ClientIMPL) metricsRequest(ctx context.Context, response interface{}, e
...
+func (c *ClientIMPL) mirrorTransferRate(ctx context.Context, response interface{}, entityID string, limit int) error {
+	qp := getFSDefaultQueryParams(c)
+	qp.RawArg("id", fmt.Sprintf("eq.%s", entityID))
+	qp.Limit(limit)
+	qp.RawArg("select", "id,timestamp,synchronization_bandwidth,mirror_bandwidth,data_remaining")
+
+	customHeader := http.Header{}
+	customHeader.Add("DELL-VISIBILITY", "Internal")
+	apiClient := c.APIClient()
+	apiClient.SetCustomHTTPHeaders(customHeader)
```
Comment (adarsh-dell) [inline metrics.go:63]: @nitesh3108 Shouldn't we unset this custom header at the last?

File: metrics.go [hunk 2]
```diff
@@ -66,6 +98,13 @@ func (c *ClientIMPL) PerformanceMetricsByVolume(ctx context.Context, entityID st
 	return resp, err
 }
 
+// VolumeMirrorTransferRate - Volume Mirror Transfer Rate
+func (c *ClientIMPL) VolumeMirrorTransferRate(ctx context.Context, entityID string) ([]VolumeMirrorTransferRateResponse, error) {
+	var resp []VolumeMirrorTransferRateResponse
+	err := c.mirrorTransferRate(ctx, &resp, entityID, 2000)
```
Comment (adarsh-dell) [inline metrics.go:104]: Can this `2000` be a constant value instead of hardcoding it here like this way?

Comment (nitesh3108) [reply metrics.go]: done

File: metrics.go (hunk 1 above)
Comment (nitesh3108) [reply metrics.go:64]: Yes, its done

**Review by adarsh-dell — DISMISSED** [review]
LGTM.

File: metrics.go [hunk 3]
```diff
@@ -45,6 +51,34 @@ func (c *ClientIMPL) metricsRequest(ctx context.Context, response interface{}, e
...
+		RequestConfig{
+			Method:      "GET",
+			Endpoint:    mirrorURL,
+			QueryParams: qp,
+		},
+		response)
+	if err != nil {
+		err = WrapErr(err)
+	}
+	customHeader.Del("DELL-VISIBILITY")
```
Comment (AkshaySainiDell) [inline metrics.go:77]:
```suggestion
	customHeader.Del("DELL-VISIBILITY")
	apiClient.SetCustomHTTPHeaders(customHeader)
```

**Review by AkshaySainiDell — CHANGES_REQUESTED** [review]

**Review by adarsh-dell — DISMISSED** [review]

File: metrics.go (hunk 3 above)
Comment (nitesh3108) [reply metrics.go:77]: Done

**Review by adarsh-dell — APPROVED** [review]

**Review by AkshaySainiDell — APPROVED** [review]

**Pull request merged**